heracles/dices/jackknife.py

```diff
@@ -327,15 +327,6 @@ def delete2_correction(cls0, cls1, cls2):
             Q_ii.append(qii)
     # Compute the correction from the ensemble
     Q = _jackknife_covariance(Q_ii, nd=2)
-    # Diagonalise the correction
-    for key in Q:
-        q = Q[key]
-        *_, length = q.shape
-        q_diag = np.diagonal(q, axis1=-2, axis2=-1)
-        q_diag_exp = np.zeros_like(q)
-        diag_indices = np.arange(length)  # Indices for the diagonal
-        q_diag_exp[..., diag_indices, diag_indices] = q_diag
-        Q[key] = Result(q_diag_exp, axis=q.axis, ell=q.ell)
```

Comment:
Why is this no longer necessary here? Doesn't the change also change the result of this function?

Reply:
Indeed it does but I think this way it is better. Rather than diagonalizing the correction and then simply adding it to the cov_jk, we can save the fully correction (in case somebody wishes to have a look) and then substract its diagonal to the diagonal of the cov_jk.